src/web/js/check-ui.js

```diff
@@ -297,220 +477,90 @@
-                          if(test_index === 0) {
-                            block.on("click", ".check-block-header",
-                                     function(e) {
-                                       if(block.hasClass("expanded"))
-                                         wrapper.trigger('click');
-                                     });
-                          }
-                          block.one("click", ".check-block-header", function(e) {
-                            if (currentlyExpandedBlock === block && neverOpened){
-                              outputUI.runMarks();
-                              politely(block.find(".cm-future-snippet"),
-                                       function(snip){snip.cmrefresh();});
-                              neverOpened = false;
-                            }
-                          });
-                          restarter.resume(runtime.nothing);
-                        });
-                      } else {
-                        console.error("ARRRGGH BAD THINGS HAPPENED");
-                        restarter.resume(runtime.nothing);
-                      }
-                    })
-                });
+
+      container.append($(checkResultsContainer).prepend(summary));
+      
+      }catch(e){console.error(e);}
+      
+      let maybeLocToAST   = outputUI.makeMaybeLocToAST(runtime, documents, srcloc);
+      let srclocAvaliable = outputUI.makeSrclocAvaliable(runtime, documents, srcloc);
+      let maybeStackLoc   = runtime.makeFunction(function(n, userFramesOnly) {return runtime.ffi.makeNone();});
+      
+      function vivifySkeletonFallback(restarter, skeleton) { 
+        return runtime.runThunk(function() { 
+          return get(skeleton.renderable, "render-reason").app(); 
+        }, function(result) {
+          if (runtime.isSuccessResult(result)) {
+            runtime.runThunk(function() {
+              return outputUI.renderErrorDisplay(documents, runtime, result.result, []);
+            }, function(result) {
+              if (runtime.isSuccessResult(result)) {
+                if(skeleton.pyretStack)
+                  result.result.append(outputUI.renderStackTrace(runtime, documents, srcloc, skeleton.pyretStack));
+                skeleton.vivify(result.result);
+                restarter.resume(runtime.nothing);
               } else {
-                var snippet =
-                  outputUI.snippet(editors,
-                                   outputUI.cmPosFromSrcloc(runtime, srcloc, get(test,"loc")))
-                  .wrapper;
-                reason.replaceWith(snippet);
-                block.one("click", ".check-block-header",
-                          function(e) {
-                            snippet[0].cmrefresh();
-                          });
+                skeleton.vivify(
+                  $("<span>").text("An error occurred rendering the reason for this error; details logged to the console"));
+                console.error(result);
```

Comment:
Probably worth adding some informative text to this console statement, to help distinguish it from the one on line 512?